5.findBugCommit.js

```diff
@@ -23,9 +23,35 @@
 //    index: 78
 //  }  
 
+// sorted = binary 
+// find mid idx = split in half --> think phonebook 
+// should have left side and right side 
+// left should start 0
+// right is end so arr.len -1 
+// if target === midIdx then return that 
+// if midIdx is less then 
+// [1, 2, 3, 4, 5]
+// mid is 3, lets say we're looking for 5 so if mid if less than 5
+// we know we need to go to the right 
+// if mid is greater, we go to the left (might be confusing left and right?)
+
+
 const identifyFirstBadCommit = (commits) => {
-  
+    let left = 0;
+    let right = commits.length - 1
+    while(left <= right) {
+        let middleIdx  = Math.floor((left + right) / 2);
+        if(commits[middleIdx] === middleIdx) {
+            return middleIdx
+        } else if(commits[middleIdx] < middleIdx){
+            // we want to go thru the right side 
+        return middleIdx + 1
+        } else {
+            return middleIdx - 1
+        }
+    }
+
```

Comment:
returning in while loop is the end of the function. If you're returning for all cases, there would be no point to looping. Also, the problem is asking for an object as the output.